frontend/public/components/environment.jsx

```diff
@@ -80,6 +80,149 @@ const stateToProps = ({k8s}, {obj}) => ({
   model: k8s.getIn(['RESOURCES', 'models', referenceFor(obj)]) || k8s.getIn(['RESOURCES', 'models', obj.kind]),
 });
 
+class CurrentEnvVars {
+  constructor() {
+    this.currentEnvVars = {};
+  }
+
+  instantiateWithRawData(rawEnvData) {
```

Comment:
It's not clear to me what "raw data" is? Is that the original object?

Reply:
Maybe `setRawData`?

Reply:
I used to call it setRawData, but I wanted to separate how we create a full instance of the object vs a set of a specific property.

Reply:
"Instantiate" seems like the wrong verb because the object is already created and this isn't a constructor. Maybe `initRawData`?

But I'm not sure I'm convinced that's better than `setRawData` :)